### PR TITLE
Support QuestManager on Creatures

### PR DESCRIPTION
**Pull request by LtRipley36706**

### CODE REVIEW

**Comment by ziang4891** [comment]
Tested working on my local server

File: Source/ACE.Server/Managers/QuestManager.cs [hunk 1]
```diff
@@ -20,8 +20,38 @@ public class QuestManager
 
         public Player Player { get; }
```
Comment (gmriggs) [inline Source/ACE.Server/Managers/QuestManager.cs]: there doesn't need to be separate fields for Player and Creature

File: Source/ACE.Server/WorldObjects/Managers/EmoteManager.cs [hunk 2]
```diff
@@ -1218,6 +1209,124 @@ public float ExecuteEmote(BiotaPropertiesEmote emoteSet, BiotaPropertiesEmoteAct
             return delay;
         }
 
+        private static void CheckForQuestManager(Creature creature)
```
Comment (gmriggs) [inline Source/ACE.Server/WorldObjects/Managers/EmoteManager.cs]: this method is unneeded. suggested pattern posted to ce channel

File: Source/ACE.Server/WorldObjects/Managers/EmoteManager.cs [hunk 3]
```diff
@@ -1218,6 +1209,124 @@ public float ExecuteEmote(BiotaPropertiesEmote emoteSet, BiotaPropertiesEmoteAct
...
+        {
+            if (creature != null && creature.QuestManager == null)
+                creature.QuestManager = new QuestManager(creature);
+        }
+
+        private void UpdateQuest(BiotaPropertiesEmoteAction emote, WorldObject targetObject, Creature creature)
```
Comment (gmriggs) [inline Source/ACE.Server/WorldObjects/Managers/EmoteManager.cs]: EmoteManager should not contain special methods for particular emotes... if these needed to be methods, they belong in QuestManager

However, there is no need for these to be separate methods

**Pull request closed without merging**